core/app/models/spree/calculator/flat_rate.rb

```diff
@@ -8,10 +8,12 @@ class Calculator::FlatRate < Calculator
     preference :currency, :string, default: ->{ Spree::Config[:currency] }
 
     def compute(object = nil)
+      return BigDecimal(0) if object.nil?
```

Comment:
I don't think this is necessary since when `object` is `nil` we still end up in the second part of the conditional below. This just causes us to check that twice.